src/sdk/namespace_fs.js

```diff
@@ -1759,6 +1761,10 @@ class NamespaceFS {
             throw new RpcError('BAD_REQUEST', 'Bad Request');
         }
     }
+
+    _skip_version_dir(dir_key) {
+        return dir_key.indexOf(`${HIDDEN_VERSIONS_PATH}`) >= 0;
```

Comment:
seems like the string interpolation (`${x}`) is unneeded 
```suggestion
        return dir_key.indexOf(HIDDEN_VERSIONS_PATH) >= 0;
```

Reply:
shouldn't we search for /.versions/ ? instead of .versions

Reply:
Yes. In the previous code, the file parts were divided with `/` delimeter and here it should be `/.versions/`. I will correct it

Reply:
Let's take for example the following dir:
dir_key = a/b/c/d/e/f/g/h/i/j/k/l/m/n/o/p/.versions
Definitions: 
n = dir_key.length
l = number of '/'
m = avg number of chars of every subdir
n = l + (l+1)*m

Option1 -
1. Splitting by '/' is 1 loop - O(n) - while n is the number of chars in dir_key.
2. Iterating the result array and comparing char by char of each item in the array O((l+1)*m) while m is the average number of chars of each item in the array and l is the number of '/' in the string = the number of items in the split result array.
Total complexity time - O(n+(l+1)*m)

Option 2 - 
1. searching char by char twice will take - O(2n)

So the question is (l+1)*m < n, the answer is yes and your assumption is correct, the difference is exactly l because we said that n = l + (l+1)*m
I think that it's a very slight improvement, I want to believe that l is not a big number in most of the cases unless I did a mistake in the calculation/ the algorithm of option 1, please correct me if you see a mistake.

Anyway, there is another optimization you can do for option 2 in order to iterate the string once - for example:
```
const idx = dir_key.indexOf(HIDDEN_VERSIONS_PATH);
return idx === 0 || idx > 0 && dir_key[idx-1] === ‘/’;
```
then the optimized version of option 2 is O(n).

Reply:
This is okay :)

Reply:
@v-harihar do we have this version path like "dir_key = a/b/c/d/e/f/g/h/i/j/k/l/m/n/o/p/.versions" only?

If yes, why can't we first split this string, with "/" as delimiter, and then just compare last element with ".versions"?

I mean to say is that if we have any idea of the position of ".versions" then compare that only.

I know you would have thought of this but just mentioning.

Reply:
So Vinayak is saying that dir_key can be exactly `.versions` (without '/') or `a/b/.versions` for example
I thought that we can use dir_key.endsWith('.versions') but still we need to check for the / if the index > 0 (that's why I wrote indexOf and not endsWith) / compare to .versions or search for dir_key.lastIndexOf('/') and then slice and compare to .versions

Reply:
@aspandey Yes logic wise both `split & compare` and the `indexOf() & check index (the added logic)` would work fine.  Also process cycle wise both results same. We had that logic earlier in this patch and overwritten because of append push which I should not have done :)